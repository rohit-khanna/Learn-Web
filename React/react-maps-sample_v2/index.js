import React from "react";
import { render } from "react-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import "./index.css";
import App from "./src/components/App";
import ConfigureStore from "./src/redux/configureStore";
import InitialState from "./src/redux/reducers/InitialState";
import { Provider as ReduxProvider } from "react-redux";

const store = new ConfigureStore(InitialState);

render(
  <ReduxProvider store={store}>
    <App />
  </ReduxProvider>,
  document.getElementById("app")
);
