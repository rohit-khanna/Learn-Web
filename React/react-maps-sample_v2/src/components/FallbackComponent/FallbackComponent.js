import React from "react";

/**
 * Fallback component, in case of unexpected error from any React Element
 * @param {*} param0
 */
const FallbackComponent = ({ errorInfo, error }) => {
  return (
    <div className="alert alert-danger">
      <div className="alert-heading">We&apos;ve got some problem</div>
      <p>{error}</p>
      <details>{JSON.stringify(errorInfo)}</details>
    </div>
  );
};

export default FallbackComponent;
