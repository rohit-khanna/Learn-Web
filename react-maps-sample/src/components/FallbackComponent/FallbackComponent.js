import React from "react";
import Alert from "react-bootstrap/Alert";

/**
 * Fallback component, in case of unexpected error from any React Element
 * @param {*} param0 
 */
const FallbackComponent = ({ errorInfo, error }) => {
  return (
    <Alert variant="danger">
      {" "}
      <Alert.Heading>We&apos;ve got some problem</Alert.Heading>
      <p>{error}</p>
      <details>{JSON.stringify(errorInfo)}</details>
    </Alert>
  );
};

export default FallbackComponent;
