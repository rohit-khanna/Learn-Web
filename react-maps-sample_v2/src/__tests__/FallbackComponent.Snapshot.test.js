import renderer from "react-test-renderer";
import React from "react";
import FallbackComponent from "../components/FallbackComponent";

describe("Fallback Component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(
      <FallbackComponent errorInfo="some info" error="some error" />
    );

    expect(tree).toMatchSnapshot();
  });
});
