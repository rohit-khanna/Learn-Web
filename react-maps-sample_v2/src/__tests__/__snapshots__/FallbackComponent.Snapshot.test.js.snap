// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fallback Component matches the snapshot 1`] = `
<div
  className="alert alert-danger"
>
  <div
    className="alert-heading"
  >
    We've got some problem
  </div>
  <p>
    some error
  </p>
  <details>
    "some info"
  </details>
</div>
`;
